controllers/operands/deploymentHandler.go

```diff
@@ -15,23 +15,41 @@ import (
 	"github.com/kubevirt/hyperconverged-cluster-operator/controllers/common"
 )
 
+type deploymentOperand struct {
+	operand *genericOperand
+}
+
+func (dp deploymentOperand) ensure(req *common.HcoRequest) *EnsureResult {
+	return dp.operand.ensure(req)
+}
+
+func (dp deploymentOperand) reset() { /* no implementation */ }
+
 func newDeploymentHandler(Client client.Client, Scheme *runtime.Scheme, required *appsv1.Deployment) Operand {
-	h := &genericOperand{
-		Client: Client,
-		Scheme: Scheme,
-		crType: "Deployment",
-		hooks:  &deploymentHooks{required: required},
+	h := &deploymentOperand{
+		operand: &genericOperand{
+			Client: Client,
+			Scheme: Scheme,
+			crType: "Deployment",
+			hooks:  &deploymentHooks{required: required},
+		},
 	}
-
 	return h
 }
 
 type deploymentHooks struct {
 	required *appsv1.Deployment
 }
 
-func (h deploymentHooks) getFullCr(_ *hcov1beta1.HyperConverged) (client.Object, error) {
-	return h.required.DeepCopy(), nil
+func (h deploymentHooks) getFullCr(hc *hcov1beta1.HyperConverged) (client.Object, error) {
+	switch h.required.Name {
+	case string(util.AppComponentUIPlugin):
+		return NewKvUIPluginDeployment(hc), nil
+	case string(util.AppComponentUIProxy):
+		return NewKvUIProxyDeployment(hc), nil
+	default:
+		return h.required.DeepCopy(), nil
+	}
```

Comment:
I wonder, but not sure, that we can use two different hooks here. We can still use common code for the parts that requires it.

Also, is the `default` part in use?

Reply:
you mean a different deployment hook for UI plugin and UI proxy deployments?
doesn't it mean each one should have `getFullCr()`, `getEmptyCr()`, and `updateCr()` methods? or i misunderstood how it is possible to have 2 hooks under `deploymentOperand`.

the default part is currently not being executed, unless we'll have another deployment to handle. should i remove it and put an error instead?